code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -223,16 +307,24 @@ func (msc *MinerSmartContract) nodeStatHandler(ctx context.Context,
 	params url.Values, balances cstate.StateContextI) (
 	resp interface{}, err error) {
 
-	var (
-		id = params.Get("id")
-		sn *MinerNode
-	)
+	var id = params.Get("id")
 
-	if sn, err = getMinerNode(id, balances); err != nil {
-		return nil, smartcontract.NewErrNoResourceOrErrInternal(err, true, cantGetMinerNodeMsg)
+	if balances.GetEventDB() != nil {
```

Comment:
Again we can't use balances.GetEventDB, will be fixed in  [issue 657](https://github.com/0chain/0chain/issues/657). Can't we use pre existing code.

Reply:
corrected